relayer/relay_message_api_handler.go

```diff
@@ -0,0 +1,53 @@
+package relayer
+
+import (
+	"encoding/json"
+	"math/big"
+	"net/http"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const RelayMessageApiPath = "/relay-message"
+
+type RelayMessageRequest struct {
+	// cb58 encoding of the blockchain ID
+	BlockchainID string `json:"blockchain-id"`
+	// Hex encoding of the warp message ID
+	MessageID string `json:"message-id"`
+	// Integer representation of the block number
+	BlockNum string `json:"block-num"`
```

Comment:
Off-chain Warp messages do not have a block number associated with them. We should provide a route for fully specifying a Warp message so that we don't have to fetch from the source chain at a specific height.

I'm thinking we have `/relay` that accepts a Warp message ID as well as a height, and a /relay/message` that accepts the full unsigned message bytes, but no height.

Reply:
In order to process a message in the relayer we need: `NetworkID`, `SourceAddress`, `SourceChainID`, and `Payload`. Do we want to specify all of these for this endpoint?

Reply:
The idea is to make the block number optional later, but it requires implementing an iterative lookback for on-chain messages.

Reply:
I see two options:
1. Provide a separate endpoint that fully specifies the off-chain Warp message, including the fields you mentioned
2. Implement control flow to fetch the off-chain Warp message by it's ID using the Warp API. The downside here is that the public APIs disable the Warp API.